### PR TITLE
Issue 882 - add service definition doc

### PR DESCRIPTION
**Pull request by dabooz**

### CODE REVIEW

File: doc/service_def.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Service Definition
```
Comment (linggao) [inline doc/service_def.json:1]: Should the file name be service_def.md instead of service_def.json?

File: doc/service_def.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+OpenHorizon deploys services to edge nodes, where those services are comprised of at least one container image and a configuration that conditions how the service executes.
+Services are defined by a service definition, encoded in JSON. The attributes of a service definition are defined in this document.
+
+## Structure
+
+Service defintions come in 2 forms;
+- `source`: A service definition file that is part of the source code in a service project, e.g. https://github.com/open-horizon/examples/blob/master/edge/services/helloworld/horizon/service.definition.json.
+In this case, the service definition is used to create or update a service definition in the OpenHorizon Exchange.
+- `display`: A service definition displayed by `hzn exchange service list`.
+In this case, the service definition as extra fields and some fields have a different datatype as compared to the source form version of a service definition.
```
Comment (linggao) [inline doc/service_def.json:12]: as -> has

File: doc/service_def.json (hunk 1 above)
Comment (dabooz) [reply doc/service_def.json:1]: ah, good one. I have no idea why I named it .json, that's just wrong.

File: doc/service_def.json (hunk 2 above)
Comment (dabooz) [reply doc/service_def.json:12]: fixed

**Pull request merged**